src/Xamarin.Android.Build.Tasks/Tasks/ResolveSdksTask.cs

```diff
@@ -134,6 +126,35 @@ public override bool Execute ()
 			//note: this task does not error out if it doesn't find all things. that's the job of the targets
 			return !Log.HasLoggedErrors;
 		}
+
+		string GetJvmPath ()
+		{
+			var key = new Tuple<string, string> (nameof (ResolveSdks), JavaSdkPath);
+			var cached = BuildEngine4.GetRegisteredTaskObject (key, RegisteredTaskObjectLifetime.AppDomain) as string;
```

Comment:
@jonathanpeppers is `AppDomain` the right thing we need here?

Reply:
Yeah, this part looks right, using `AppDomain` allows VS to reuse the value across builds. If `JavaSdkPath` changes, since it's part of the key, it would find the path again.